jupyter_server/services/contents/largefilemanager.py

```diff
@@ -18,7 +18,7 @@ def save(self, model, path=""):
         if chunk is not None:
             path = path.strip("/")
 
-            self.run_pre_save_hook(model=model, path=path)
+            self.run_pre_save_hooks(model=model, path=path)
```

Comment:
In looking at the history of this issue, including https://github.com/jupyterlab/jupyterlab/pull/11599, it led to PR https://github.com/jupyter-server/jupyter_server/pull/643, where this call _was_ within the [`chunk == 1` block](https://github.com/jupyter-server/jupyter_server/pull/643/files#diff-25c728260b9f71e77e54ba095a355509e4d0cc0d57f7b6242d181569f225975bR39-L38).  As a result, that PR winds up calling the pre-save hook on _every_chunk, although the post-save hook is (still) only called on the last chunk.

I suspect a bug was introduced there but wanted to bring it up here.

Reply:
Yes, I agree.

Reply:
I think this particular hook needs to still be constrained to chunk 1 even though chunks don't appear to be in use (and handled in the `else` block).

Reply:
Yes, but it's part of #716. How does it sound to merge this PR and then #716?

Reply:
I'm fine with merging #716 first :+1: